modules/cloud-function-v2/main.tf

```diff
@@ -24,28 +24,17 @@ locals {
       : null
     )
   )
-  _iam_run_invoker_members = concat(
-    lookup(var.iam, "roles/run.invoker", []),
-    var.trigger_config == null ? [] :
-    var.trigger_config.service_account_create ? ["serviceAccount:${local.trigger_service_account_email}"] : []
-  )
-  iam = merge(
-    var.iam,
-    length(local._iam_run_invoker_members) == 0 ? {} :
-    {
-      "roles/run.invoker" : local._iam_run_invoker_members
-    },
-  )
   prefix = var.prefix == null ? "" : "${var.prefix}-"
   service_account_email = (
     var.service_account_create
     ? google_service_account.service_account[0].email
     : var.service_account
   )
-  trigger_service_account_email = (
-    try(var.trigger_config.service_account_create, false)
-    ? google_service_account.trigger_service_account[0].email
-    : try(var.trigger_config.service_account_email, null)
+  trigger_sa_create = (
+    try(var.trigger_config.service_account_create, false) == true
```

Comment:
why do we need == true?

Reply:
because it could be passed in as `null`, pass through the `try`, and raise a type error as it would not be a boolean